### PR TITLE
additional files for release zip

### PR DESCRIPTION
**Pull request by haofan-ms**
Update: additional cloud provider log based on PR https://github.com/Azure/aks-engine/pull/3792

### CODE REVIEW

File: diagnosis/README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ The logs retrieved by `getkuberneteslogs.sh` are the following:
 - etcd status and journal
 - docker status and journal
 - kube-system snapshot
+- Kubernetes cloud provider logs
```
Comment (jadarsie) [inline diagnosis/README.md:22]: this is more like `Azure CNI config files`

File: diagnosis/scripts/release.sh [hunk 2]
```diff
@@ -4,5 +4,5 @@ VERSION=$1
 
 dos2unix *.sh
 mkdir -p _out
-tar -czf _out/diagnosis-v${VERSION}.tar.gz collectlogs.sh getkuberneteslogs.sh hosts.sh
-zip -q _out/diagnosis-v${VERSION}.zip collectlogs.sh getkuberneteslogs.sh hosts.sh
+tar -czf _out/diagnosis-v${VERSION}.tar.gz collectlogs.sh getkuberneteslogs.sh hosts.sh azs-collect-windows-logs.ps1
```
Comment (jadarsie) [inline diagnosis/scripts/release.sh:7]: are the ps1 files consumed?

Comment (haofan-ms) [reply diagnosis/scripts/release.sh:7]: Yes, https://github.com/msazurestackworkloads/azurestack-gallery/blob/master/diagnosis/getkuberneteslogs.sh#L136

File: diagnosis/README.md (hunk 1 above)
Comment (haofan-ms) [reply diagnosis/README.md]: updated

File: diagnosis/k8sCollectLogsCi.sh [hunk 3]
```diff
@@ -200,32 +188,42 @@ then
     scp ${SCP_FLAGS} hosts.sh ${USER}@${MASTER_IP}:/home/${USER}/hosts.sh
     ssh ${SSH_FLAGS} ${USER}@${MASTER_IP} "sudo chmod 744 hosts.sh; ./hosts.sh"
     scp ${SCP_FLAGS} ${USER}@${MASTER_IP}:/home/${USER}/cluster-snapshot.zip ${LOGFILEFOLDER}/cluster-snapshot.zip
-    ssh ${SSH_FLAGS} ${USER}@${MASTER_IP} "kubectl get nodes -o json|jq -r '.items[].metadata.name' | grep -v  k8s-master-* > cluster_nodes.txt"
+    ssh ${SSH_FLAGS} ${USER}@${MASTER_IP} $'kubectl get nodes -o json|jq -r \'.items[] | select(.metadata.labels."kubernetes.io/os" == "linux") | .metadata.name\' > linux_nodes.txt'
```
[the remark this answers is not in the document]
Comment (haofan-ms) [reply diagnosis/k8sCollectLogsCi.sh:191]: These changes are updated to match the logic in https://github.com/msazurestackworkloads/azurestack-gallery/blob/master/diagnosis/getkuberneteslogs.sh#L303-L320.

File: diagnosis/k8sCollectLogsCi.sh [hunk 4]
```diff
@@ -24,6 +24,17 @@ processHost()
...
 }
 
+processWindowsHost()
+{
+    host=$1
+
+    echo "[$(date +%Y%m%d%H%M%S)][INFO] Processing windows-host ${host}"
+    scp ${SCP_FLAGS} -o ProxyCommand="${PROXY_CMD}" azs-collect-windows-logs.ps1 ${USER}@${host}:"C:/k/debug/azs-collect-windows-logs.ps1"
+    ssh ${SSH_FLAGS} -o ProxyCommand="${PROXY_CMD}" ${USER}@${host} "powershell; Start-Process PowerShell -Verb RunAs; C:/k/debug/azs-collect-windows-logs.ps1"
+    scp ${SCP_FLAGS} -o ProxyCommand="${PROXY_CMD}" ${USER}@${host}:"C:/Users/azureuser/win_log_${host}.zip" ${LOGFILEFOLDER}/"win_log_${host}.zip"
```
Comment (jadarsie) [inline diagnosis/k8sCollectLogsCi.sh:34]: do not assume `azureuser`

Comment (haofan-ms) [reply diagnosis/k8sCollectLogsCi.sh]: Updated

File: diagnosis/k8sCollectLogsCi.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (haofan-ms) [reply diagnosis/k8sCollectLogsCi.sh]: Updated

File: diagnosis/k8sCollectLogsCi.sh [hunk 5]
```diff
@@ -24,6 +24,17 @@ processHost()
...
 
+processWindowsHost()
+{
+    host=$1
+
+    echo "[$(date +%Y%m%d%H%M%S)][INFO] Processing windows-host ${host}"
+    scp ${SCP_FLAGS} -o ProxyCommand="${PROXY_CMD}" azs-collect-windows-logs.ps1 ${USER}@${host}:"C:/k/debug/azs-collect-windows-logs.ps1"
+    ssh ${SSH_FLAGS} -o ProxyCommand="${PROXY_CMD}" ${USER}@${host} "powershell; Start-Process PowerShell -Verb RunAs; C:/k/debug/azs-collect-windows-logs.ps1"
+    scp ${SCP_FLAGS} -o ProxyCommand="${PROXY_CMD}" ${USER}@${host}:"C:/Users/${USER}/win_log_${host}.zip" ${LOGFILEFOLDER}/"win_log_${host}.zip"
+    ssh ${SSH_FLAGS} -o ProxyCommand="${PROXY_CMD}" ${USER}@${host} "powershell; rm C:/k/debug/azs-collect-windows-logs.ps1; rm C:/Users/azureuser/win_log_${host}.zip"
```
Comment (jadarsie) [inline diagnosis/k8sCollectLogsCi.sh:35]: `azureuser` here as well

File: diagnosis/k8sCollectLogsCi.sh [hunk 6]
```diff
@@ -200,32 +188,39 @@ then
...
-    scp ${SCP_FLAGS} ${USER}@${MASTER_IP}:/home/${USER}/cluster_nodes.txt ${LOGFILEFOLDER}/cluster-nodes.txt
-    ssh ${SSH_FLAGS} ${USER}@${MASTER_IP} "sudo rm -f cluster-snapshot.zip hosts.sh cluster_nodes.txt"
+    scp ${SCP_FLAGS} ${USER}@${MASTER_IP}:/home/${USER}/*_nodes.txt ${LOGFILEFOLDER}/.
+    ssh ${SSH_FLAGS} ${USER}@${MASTER_IP} "sudo rm -f cluster-snapshot.zip hosts.sh *_nodes.txt"
     
-    if [ ! -f ${LOGFILEFOLDER}/cluster-nodes.txt ]
+    if [ ! -f ${LOGFILEFOLDER}/linux_nodes.txt ]
     then
-        echo "Agent nodes not present"
+        echo "Linux nodes not present"
```
Comment (jadarsie) [inline diagnosis/k8sCollectLogsCi.sh:196]: not sure how useful is to print this message

File: diagnosis/k8sCollectLogsCi.sh [hunk 7]
```diff
@@ -200,32 +188,39 @@ then
...
+        
+        for host in ${WINDOWS_NODES}
+        do
+            processHost ${host}
+        done
+    fi
 fi
 
-echo "[$(date +%Y%m%d%H%M%S)][INFO] Logs can be found here: $LOGFILEFOLDER"
+echo "[$(date +%Y%m%d%H%M%S)][INFO] Done with k8s log collection"
```
Comment (jadarsie) [inline diagnosis/k8sCollectLogsCi.sh:225]: it may be useful to print the output directory

File: diagnosis/k8sCollectLogsCi.sh (hunk 6 above)
Comment (haofan-ms) [reply diagnosis/k8sCollectLogsCi.sh:196]: I feel in cases we are not getting logs from agent nodes, it help identify if it's configuration issue or log-collection issue. Does this make sense?

File: diagnosis/k8sCollectLogsCi.sh (hunk 7 above)
Comment (haofan-ms) [reply diagnosis/k8sCollectLogsCi.sh]: Updated

**Review by jadarsie — APPROVED** [review]

**Pull request merged**